src/Language/PureScript/Errors.hs

```diff
@@ -261,6 +248,31 @@ data LabelType = TypeLabel | SkolemLabel String deriving (Show, Eq, Ord)
 -- | A map from rigid type variable name/unknown variable pairs to new variables.
 type UnknownMap = M.Map (LabelType, Unknown) Unknown
 
+-- |
+-- Extract nested error messages from wrapper errors
+--
+unwrapErrorMessage :: ErrorMessage -> Maybe ErrorMessage
+unwrapErrorMessage em = case em of
+  (ErrorCheckingKind _ err)       -> Just err
```

Comment:
This isn't going to work for certain errors. You need to recursively call `unwrapErrorMessage` on the wrapped error until you get down to an unwrapped error. On the plus side, this means you can get rid of the `Maybe`.